client/task_runner_test.go

```diff
@@ -397,3 +398,47 @@ func TestTaskRunner_Validate_UserEnforcement(t *testing.T) {
 	}
 
 }
+
+func TestTaskRunnerResouseUsageTS(t *testing.T) {
+	_, tr := testTaskRunner(false)
+
+	t1, _ := time.Parse(time.RFC822, "02 Jan 06 15:03 MST")
+	t2, _ := time.Parse(time.RFC822, "02 Jan 06 15:05 MST")
+	t3, _ := time.Parse(time.RFC822, "02 Jan 06 15:06 MST")
+	t4, _ := time.Parse(time.RFC822, "02 Jan 06 15:07 MST")
+	t5, _ := time.Parse(time.RFC822, "02 Jan 06 15:08 MST")
+
+	ru1 := cstructs.TaskResourceUsage{Timestamp: t1}
+	ru2 := cstructs.TaskResourceUsage{Timestamp: t2}
+	ru3 := cstructs.TaskResourceUsage{Timestamp: t3}
+	ru4 := cstructs.TaskResourceUsage{Timestamp: t4}
+	ru5 := cstructs.TaskResourceUsage{Timestamp: t5}
+
+	tr.resourceUsage.Enqueue(&ru1)
+	tr.resourceUsage.Enqueue(&ru2)
+	tr.resourceUsage.Enqueue(&ru3)
+	tr.resourceUsage.Enqueue(&ru4)
+	tr.resourceUsage.Enqueue(&ru5)
+
+	values := tr.ResourceUsageTS(t3.Add(-1 * time.Second))
+	if len(values) != 3 {
+		t.Fatalf("expected values: 3, actual: %v", len(values))
+	}
+
+	values = tr.ResourceUsageTS(t3)
+	if len(values) != 3 {
+		t.Fatalf("expected values: 3, actual: %v", len(values))
+	}
+
+	begenning, _ := time.Parse(time.RFC822, "01 Jan 1970 00:00 UTC")
+	values = tr.ResourceUsageTS(begenning)
+	if len(values) != 5 {
+		t.Fatalf("expected values: 5, actual: %v", len(values))
+	}
+
+	values = tr.ResourceUsageTS(t5.Add(1 * time.Second))
+	if len(values) != 0 {
+		t.Fatalf("expected values: 3, actual: %v", len(values))
```

Comment:
Either the error messages are wrong are the check :)

Also can we add a case for when the users passes `since=0`